src/main/java/org/opentripplanner/model/Timetable.java

```diff
@@ -330,14 +328,14 @@ public Result<TripTimesPatch, UpdateError> createUpdatedTripTimes(
       }
     }
 
-    OptionalInt invalidStopIndex = newTimes.findFirstNoneIncreasingStopTime();
-    if (invalidStopIndex.isPresent()) {
+    var maybeError = newTimes.validateNonIncreasingTimes();
+    if (maybeError.isFailure()) {
       LOG.debug(
         "TripTimes are non-increasing after applying GTFS-RT delay propagation to trip {} after stop index {}.",
         tripId,
-        invalidStopIndex.getAsInt()
+        maybeError.failureValue().stopIndex()
       );
-      return Result.failure(new UpdateError(feedScopedTripId, NON_INCREASING_TRIP_TIMES));
+      return (Result<TripTimesPatch, UpdateError>) maybeError;
```

Comment:
We need to rewrap the error in the correct result type here, instead of casting it

Reply:
Changed it.

Reply:
Thanks, looks good.